[DOC-9564, DOC-9585] Update docs on connecting to a cluster with no current downloads available

mdlinville opened the pull request:
1. [DOC-9564] Update docs on connecting to a cluster with no current downloads available
2. [DOC-9585] Update Connect Modal Docs to 
    1. bring up-to-date with latest UI 
    2. simplify directions to decouple from UI updates


### Previews
[src/current/releases/index.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/releases/)
[src/current/releases/v23.2.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/releases/v23.2/)
[src/current/cockroachcloud/quickstart-trial-cluster.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/cockroachcloud/quickstart-trial-cluster.html)
[src/current/cockroachcloud/deploy-a-python-to-do-app-with-flask-kubernetes-and-cockroachcloud.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/cockroachcloud/deploy-a-python-to-do-app-with-flask-kubernetes-and-cockroachcloud.html)
[src/current/cockroachcloud/connect-to-a-serverless-cluster.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/cockroachcloud/connect-to-a-serverless-cluster.html)
[src/current/cockroachcloud/connect-to-your-cluster.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/cockroachcloud/connect-to-your-cluster.html)
[src/current/v23.2/install-cockroachdb-linux.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/v23.2/install-cockroachdb-linux.html)
[src/current/v23.2/install-cockroachdb-mac.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/v23.2/install-cockroachdb-mac.html)
[src/current/v23.2/install-cockroachdb-windows.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/v23.2/install-cockroachdb-windows.html)
[src/current/v23.1/install-cockroachdb-linux.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/v23.1/install-cockroachdb-linux.html)
[src/current/v23.1/install-cockroachdb-mac.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/v23.1/install-cockroachdb-mac.html)
[src/current/v23.1/install-cockroachdb-windows.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/v23.1/install-cockroachdb-windows.html)
[src/current/v22.2/install-cockroachdb-linux.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/v22.2/install-cockroachdb-linux.html)
[src/current/v22.2/install-cockroachdb-mac.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/v22.2/install-cockroachdb-mac.html)
[src/current/v22.2/install-cockroachdb-windows.md](https://deploy-preview-18249--cockroachdb-docs.netlify.app/docs/v22.2/install-cockroachdb-windows.html)

mdlinville commented on src/current/v23.2/install-cockroachdb-mac.md:
```diff
@@ -30,6 +30,10 @@ docs_area: deploy
 Use one of the options below to install CockroachDB.
 
+{{site.data.alerts.callout_success}}
+To connect to a CockroachDB cluster running any supported version of CockroachDB from your local system, you can download and install any available version of either the SQL Shell binary or the full binary for your operating system and architecture. Refer to [Releases](https://cockroachlabs.com/docs/releases/).
```
Here I say "download and install any available version" because all the macOS downloads are experimental.

mdlinville commented on src/current/v23.2/install-cockroachdb-windows.md:
```diff
@@ -19,6 +19,10 @@ docs_area: deploy
 Use one of the options below to install CockroachDB.
 
+{{site.data.alerts.callout_success}}
+To connect to a CockroachDB cluster running any supported version of CockroachDB from your local system, you can download and install any available version of either the SQL Shell binary or the full binary for your operating system and architecture. Refer to [Releases](https://cockroachlabs.com/docs/releases/).
```
Here I say "download and install any available version" because all the Windows downloads are experimental.

celiala commented on src/current/_includes/cockroachcloud/download-the-binary.md:
```diff
@@ -1,21 +1,29 @@
     {% include_cached copy-clipboard.html %}
     ~~~ shell
-    curl https://binaries.cockroachdb.com/cockroach-{{ page.release_info.version }}.darwin-10.9-amd64.tgz | tar -xJ && sudo cp -i cockroach-{{ page.release_info.version }}.darwin-10.9-amd64/cockroach /usr/local/bin/
+    curl https://get.cockroachdb.com/dl/cockroach-latest.darwin-10.9-amd64.tgz | tar -xJ && sudo cp -i cockroach-latest.darwin-10.9-amd64/cockroach /usr/local/bin/
```
The exact command is

```
curl -L https://get.cockroachdb.com/dl/cockroach-latest.darwin-10.9-amd64.tgz | tar -xz; sudo cp -i cockroach-*/cockroach /usr/local/bin/
```

![image](https://github.com/cockroachdb/docs/assets/3051672/0ba789c4-38f5-41dd-a853-90e56319e858)

(potentially for another day/out of scope here, but the Serverless instructions deal with this by not duplicating the command in the docs, and instead leaving the steps pretty high-level:
https://www.cockroachlabs.com/docs/cockroachcloud/connect-to-a-serverless-cluster#connect-to-your-cluster)

mdlinville replied:
Yes, I think bringing Dedicated and Serverless content into sync is out of sync for this effort.

Aha, I missed the `-L`, which is needed for `curl` to follow redirects.

mdlinville replied:
Now rewritten to avoid providing the curl commands directly in docs.

celiala commented on src/current/_includes/cockroachcloud/download-the-binary.md:
```diff
@@ -1,21 +1,29 @@
     ~~~
 
+    {{site.data.alerts.callout_success}}
+    To download the ARM 64-bit architecture instead of the Intel 64-bit binary, replace all instances of `darwin-10.9-amd64` with `darwin-11.0.arm64` before running the command.
```
this doesn't work(?)

Works (you can try this from a browser):
https://get.cockroachdb.com/dl/cockroach-latest.darwin-10.9-amd64.tgz

Doesn't work (I get a Not found):
https://get.cockroachdb.com/dl/cockroach-latest.darwin-11.0.arm64.tgz

(let's leave this out then?)

mdlinville replied:
How come this doesn't work for the ARM binaries? What are ARM clients supposed to do to get the latest binary?

celiala replied:
🤦🏼 I added a dot instead of a dash. 

this works:
https://get.cockroachdb.com/dl/cockroach-latest.darwin-11.0-arm64.tgz

celiala replied:
> `darwin-10.9-amd64` with `darwin-11.0.arm64`

oh so then instead of `darwin-11.0.arm64` it should be `darwin-11.0-arm64`

mdlinville replied:
Now that I am not exposing the curl commands, this is not an issue (I suppose these will be exposed in the modal when RE-565 is addressed. Thanks for confirming that the ARM downloads do download in the browser.

rail replied:
The URL is incorrect, need to replace one of the dots to a dash:
https://get.cockroachdb.com/dl/cockroach-latest.darwin-11.0-arm64.tgz

celiala commented on src/current/_includes/cockroachcloud/download-the-binary.md:
```diff
@@ -1,21 +1,29 @@
     {% include_cached copy-clipboard.html %}
     ~~~ shell
-    curl https://binaries.cockroachdb.com/cockroach-{{ page.release_info.version }}.linux-amd64.tgz | tar -xz && sudo cp -i cockroach-{{ page.release_info.version }}.linux-amd64/cockroach /usr/local/bin/
+    curl https://get.cockroachdb.com/dl/cockroach-latest.linux-amd64.tgz | tar -xz && sudo cp -i cockroach-latest.linux-amd64/cockroach /usr/local/bin/
```
(same as previous note)
UI shows `cockroach-*/cockroach` instead of `cockroach-latest.linux-amd64/cockroach`.

celiala replied:
i.e. even though command is
`curl https://get.cockroachdb.com/dl/cockroach-latest.linux-amd64.tgz`
what actually gets downloaded is a versioned download, e.g.
- `cockroach-v23.1.14.darwin-10.9-amd64` or `cockroach-v23.1.14.linux-amd64`

so `sudo cp -i cockroach-latest.linux-amd64/cockroach` will not work because `cockroach-latest.linux-amd64` will not exist..

mdlinville replied:
That's really really difficult for me to make work with these cURL comands, because I have no way in the docs to know what version they will actually get. Let me play around with cURL for a minute.

mdlinville replied:
I worked around this by bringing Dedicated content in line with Serverless.

mdlinville replied:
Then Celia showed me that the Dedicated modal is way different. Latest commits fix the Dedicated content to match reality.

celiala commented on src/current/_includes/cockroachcloud/download-the-binary.md:
```diff
@@ -24,7 +32,7 @@
     {% include_cached copy-clipboard.html %}
     ~~~ shell
-    $ErrorActionPreference = "Stop"; [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12;$ProgressPreference = 'SilentlyContinue'; $null = New-Item -Type Directory -Force $env:appdata/cockroach; Invoke-WebRequest -Uri https://binaries.cockroachdb.com/cockroach-{{ page.release_info.version }}.windows-6.2-amd64.zip -OutFile cockroach.zip; Expand-Archive -Force -Path cock ...
+    $ErrorActionPreference = "Stop"; [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12;$ProgressPreference = 'SilentlyContinue'; $null = New-Item -Type Directory -Force $env:appdata/cockroach; Invoke-WebRequest -Uri https://get.cockroachdb.com/dl/cockroach-latest.windows-6.2-amd64.zip -OutFile cockroach.zip; Expand-Archive -Force -Path cockroach.zip; Copy-Item -Force ...
```
(same)
instead of: `cockroach/cockroach-latest.windows-6.2-amd64/cockroach.exe`
should be: `cockroach/cockroach-*/cockroach.exe`

![image](https://github.com/cockroachdb/docs/assets/3051672/702b3354-1684-4f49-a024-333dc585a112)

mdlinville replied:
Worked around this by relying on the Connect modal.

celiala commented on src/current/v23.2/install-cockroachdb-linux.md:
```diff
@@ -13,18 +13,20 @@ docs_area: deploy
 CockroachDB on ARM is <b><a href="https://www.cockroachlabs.com/docs/stable/cockroachdb-feature-availability#feature-availability-phases">Generally Available</a></b> in v23.2.0 and above. For limitations specific to ARM, refer to <a href="#limitations">Limitations</a>.
 
+{{site.data.alerts.callout_success}}
+To connect to a CockroachDB cluster running any supported version of CockroachDB from your local system, you can download and install any supported version of either the SQL Shell binary or the full binary for your operating system and architecture. Refer to [Releases](https://cockroachlabs.com/docs/releases/).
```
> To connect to a CockroachDB cluster running any supported version of CockroachDB from your local system,

I'm not sure to which cluster this is referring: is the cluster i'm connecting to a Cloud Cluster, or a CRDB binary on my local system?

mdlinville replied:
Any (Linux) cluster, including on CC -- this is in the context of self-hosted Linux installs.

celiala reviewed (commented):
Thank you!
I left notes for changes (see inline)

mdlinville commented:
@celiala PTAL at the latest commit, which brings Dedicated docs in line with Serverless docs and stops materializing the cURL commands for Dedicated, so that we rely on the logic in the modal to figure out which version of the binary to download. I needed to update the Connection String and Connection Parameters sections to match, as well. Wording about private connectivity differs a bit because different options are available there for Dedicated and Serverless.

shannonbradshaw commented on src/current/cockroachcloud/connect-to-your-cluster.md:
```diff
@@ -63,29 +63,7 @@ Azure Private Link is not yet available for [CockroachDB {{ site.data.products.d
-
-    The default database is `defaultdb`. For more information, see [Default databases](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/show-databases#preloaded-databases).
-
-1. Click **Next**.
```
Why did we remove the above as part of this PR. It seems unrelated, but perhaps I'm missing something.

shannonbradshaw replied:
I don't see anything to suggest that we're changing the experience for Dedicated users in this way here: https://cockroachlabs.atlassian.net/browse/CC-26935

shannonbradshaw replied:
![Screenshot 2024-01-26 at 11 03 42 AM](https://github.com/cockroachdb/docs/assets/1447785/3fa02a9e-ad23-41c6-8714-089e3d72c013)
We seem to have dropped the instructions supporting the above ^^^

mdlinville replied:
This was my mistake -- I assumed that the Dedicated and Serverless modals were the same because I forgot about the recently-launched multi-step wizard. The Dedicated pages show where to select the database once again in recent commits.

mikeCRL commented on src/current/_data/releases.yml:
```diff
@@ -5390,8 +5390,8 @@
     For more information, refer to
-    <a href="https://www.cockroachlabs.com/docs/cockroachcloud/create-your-cluster">Create a CockroachDB Dedicated Cluster</a> or
-    <a href="https://www.cockroachlabs.com/docs/cockroachcloud/upgrade-to-v23.2">Upgrade to CockroachDB v23.2</a>.
+    [Create a CockroachDB Dedicated Cluster](https://www.cockroachlabs.com/docs/cockroachcloud/create-your-cluster) or
```
Thank you - appreciate both points!

mikeCRL commented on src/current/cockroachcloud/connect-to-your-cluster.md:
```diff
@@ -99,7 +99,12 @@ Azure Private Link is not yet available for [CockroachDB {{ site.data.products.d
   <section class="filter-content" markdown="1" data-scope="command-line">
 
-To connect to your cluster with the [built-in SQL client](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/cockroach-sql):
+You can connect to your cluster with any supported version of the full CockroachDB binary or the [built-in SQL client](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/cockroach-sql). To download the full binary and connect to a CockroachDB {{ site.data.products.dedicated }} cluster, follow these steps.
```
Should we link "supported version" to https://www.cockroachlabs.com/docs/releases/release-support-policy#current-supported-releases ?

celiala commented:
Hi @mdlinville -

I put this in our Slack thread, but also putting directly on the PR:

Some things were tricky for me to verbalize with just text, so I made a slide deck:

https://docs.google.com/presentation/d/1RsK7qlz5JZMeGC9i30IuaC3PDMNomm6yazOtep2yKhw/edit#slide=id.p

Hopefully this all makes sense. Happy to hop on a call to walk through if easier. Thanks!

mdlinville commented:
> Hi @mdlinville -
> 
> I put this in our Slack thread, but also putting directly on the PR:
> 
> Some things were tricky for me to verbalize with just text, so I made a slide deck[1]
> 
> At a high-level:
> 
> 1. The Connect Modals for Dedicated and Serverless are different enough that a verbatim copy-paste from the Serverless instructions won't map 1:1, as some UI widgets on the Serverless Connect Modals aren't there for Dedicated, and vice versa (slides 3-7 go through the differences a bit more).
> 2. I'm don't know if this feature needs to be so present in some of other pages that were updated? (slide deck points to the places)
> 3. For the app page in particular, connecting an app doesn't require the user to download a binary, so we should leave the original directions (to connect via Connection String)
> 
> [1] https://docs.google.com/presentation/d/1RsK7qlz5JZMeGC9i30IuaC3PDMNomm6yazOtep2yKhw/edit#slide=id.p
> 
> Hopefully this all makes sense. Happy to hop on a call to walk through if easier. Thanks!

I think I have addressed the feedback here and in the slides. There were some items where I had to go in a different direction and I tried to leave explanations as to why.

mdlinville commented:
> Hi @mdlinville -
> 
> I put this in our Slack thread, but also putting directly on the PR:
> 
> Some things were tricky for me to verbalize with just text, so I made a slide deck[1]
> 
> At a high-level:
> 
> 1. The Connect Modals for Dedicated and Serverless are different enough that a verbatim copy-paste from the Serverless instructions won't map 1:1, as some UI widgets on the Serverless Connect Modals aren't there for Dedicated, and vice versa (slides 3-7 go through the differences a bit more).
> 2. I'm don't know if this feature needs to be so present in some of other pages that were updated? (slide deck points to the places)
> 3. For the app page in particular, connecting an app doesn't require the user to download a binary, so we should leave the original directions (to connect via Connection String)
> 
> [1] https://docs.google.com/presentation/d/1RsK7qlz5JZMeGC9i30IuaC3PDMNomm6yazOtep2yKhw/edit#slide=id.p
> 
> Hopefully this all makes sense. Happy to hop on a call to walk through if easier. Thanks!

Done

shannonbradshaw commented on src/current/_data/releases.yml:
```diff
@@ -5390,8 +5390,8 @@
     For more information, refer to
-    <a href="https://www.cockroachlabs.com/docs/cockroachcloud/create-your-cluster">Create a CockroachDB Dedicated Cluster</a> or
-    <a href="https://www.cockroachlabs.com/docs/cockroachcloud/upgrade-to-v23.2">Upgrade to CockroachDB v23.2</a>.
+    [Create a CockroachDB Dedicated Cluster](https://www.cockroachlabs.com/docs/cockroachcloud/create-your-cluster) or
```
```suggestion

```

mdlinville replied:
Done

shannonbradshaw commented on src/current/cockroachcloud/connect-to-your-cluster.md:
```diff
@@ -63,132 +63,77 @@ Azure Private Link is not yet available for [CockroachDB {{ site.data.products.d
-1. Select a connection method (the instructions below will adjust accordingly):
+    The **Connect** page of the **Connection info** dialog displays.
+
+1. In the dialog, select the tab for a connection method, then follow the instructions below for that method.
```
The changes in lines 66 through 86 seem largely irrelevant to the product change. Why did we make these changes here?

shannonbradshaw replied:
@mdlinville ^^^

shannonbradshaw commented on src/current/cockroachcloud/connect-to-your-cluster.md:
```diff
@@ -63,132 +63,77 @@ Azure Private Link is not yet available for [CockroachDB {{ site.data.products.d
+  </section>
 
-    </section>
+  <section class="filter-content" markdown="1" data-scope="connection-string">
```
The tab choice for this data scope reads "General Connection String", but in the UI it's "Connection String".

mdlinville replied:
Done, great catch. I verified that other instances of "General connection string" are referring to the Serverless modal.

shannonbradshaw reviewed (approved):
Thanks, Matt! I did make a couple of small suggestions.

mdlinville commented on src/current/v23.2/install-cockroachdb-linux.md:
```diff
@@ -13,15 +13,13 @@ docs_area: deploy
-
 {% include cockroachcloud/use-cockroachcloud-instead.md %}
 
+<p>See <a href="https://www.cockroachlabs.com/docs/releases/{{page.version.version}}" class="mac-releasenotes-download" id="mac-releasenotes-download-{{page.version.version}}" data-eventcategory="mac-releasenotes-download">Release Notes</a> for what's new in the latest release, {{ page.release_info.version }}. To upgrade to this release from an older version, see <a href="upgrade-cockroach-versio ...
```
Whoops, this note in the Linux page has been having macOS anchors for a while. Let me fix that.

celiala reviewed (approved):
(My review is scoped to DOC-9564) This works for the release team -- thanks!
